test-cypress/integration/explore/similarity.spec.js

```diff
@@ -0,0 +1,299 @@
+describe('Similarity screen validation', function () {
+
+let indexName = 'index-' + Date.now();
+let repoId = 'graphRepo' + Date.now();
+let messageSuccess = 'Imported successfully';
+let graphToDelete = "urn:9";
+let graphToSearch = "urn:2";
+let testRepoId = 'test';
+let buildParam = " -trainingcycles 4";
+let stopWord = "stopword";
+let luceneAnalyzer = "org.apache.lucene.analysis.de.GermanAnalyzer";
+let fileToImport = "people.zip";
+let validSnippetRDFfFormat = "Turtle";
+let modifiedDataQuery = "SELECT ?documentID ?documentText { \n" +
+    "?documentID <http://dbpedia.org/ontology/birthDate> ?documentText . \n" +
+    "filter(isLiteral(?documentText)) \n" +
+    "}order by asc(str(?documentID))";
+
+let modifiedSearchQuery = "PREFIX :<http://www.ontotext.com/graphdb/similarity/> \n" +
+                          "SELECT ?documentID ?score { \n" +
+                          "?search a ?index ; \n" +
+                          "?searchType ?query; \n" +
+                          ":searchParameters ?parameters; \n" +
+                          "?resultType ?result . \n" +
+                          "?result :value ?documentID ; \n" +
+                          ":score ?score. \n" +
+                          "OPTIONAL { ?result <http://dbpedia.org/ontology/birthPlace> ?birthDate . }}";
+
+	before(function () {
+	//Sets the default timeout to 15s. so there's plenty of time for index building.
+	//defaultCommandTimeout is set per spec so it won't affect other tests.
```

Comment:
I just set a number high enough to ensure successful index building on my local machine. Please suggest how to come up with an adequate number for the timeout.

Reply:
The best approach would have been if GDB had a service which could be used to check if the index is ready. Then you could just poll the service until its done and then continue with the tests. Another option is to see if this could be done through the UI. I mean if there is a way to have some visual clue that the index is ready, you could just wait for it. These are better options and I'm sorry I can't give you the exact solution, but I'm not very familiar with the GDB services yet, so you need to consult Desi I guess.